doc/guides/debugging/probes.rst

```diff
@@ -0,0 +1,143 @@
+.. _debug-probes:
+
+Debug Probes
+############
```

Comment:
Introductory text defining a debug probe and its connections in the system would be nice to have. Something like:

```
A *debug probe* is special hardware which allows you to control execution
of a Zephyr application running on a separate board. Debug probes usually
allow reading and writing registers and memory, and support breakpoint
debugging of the Zephyr application on your host workstation using tools like
GDB. They may also support other debug software and more advanced features
such as :ref:`tracing program execution <tracing>`. For details on the related
host software supported by Zephyr, see :ref:`debug-host-tools`.

Debug probes are usually connected to your host workstation via USB; they
are sometimes also accessible via an IP network or other means. They usually
connect to the device running Zephyr using the JTAG or SWD protocols. Debug
probes are either separate hardware devices or circuitry integrated into the same
board which runs Zephyr.
```